plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -596,7 +608,16 @@ public LocatedTableHandle getTableHandle(
         boolean managed = table.get().managed();
 
         String tableLocation = table.get().location();
-        TableSnapshot tableSnapshot = getSnapshot(session, tableName, tableLocation, Optional.empty(), endVersion);
+        TableSnapshot tableSnapshot;
+        if (endVersion.isEmpty()) {
+            tableSnapshot = getCurrentSnapshot(session, tableName, tableLocation, Optional.empty());
+        }
+        else {
+            TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+            long version = getSnapshotIdFromVersion(fileSystem, tableLocation, endVersion.get());
```

Comment:
Maybe getVersionNumberFromConnectorTableVersion? Right now we get snapshotId and assign it to version.